libs/Notifications.js

```diff
@@ -76,20 +78,30 @@ function NotificationsManager(workingFolderPath, storageFolderPath) {
 		});
 	}
 
-	function buildNotification(message, timestamp, filename) {
-		return { filename, message, timeout: undefined, timestamp: timestamp ? timestamp : new Date().getTime() };
+	function buildNotification(message, timestamp, filename, ttl) {
+		return {
+			filename,
+			message,
+			timeout: undefined,
+			timestamp: timestamp ? timestamp : new Date().getTime(),
+			ttl
+		};
 	}
 
-	function addMessageToQueue(queueName, message, callback) {
-		const notificationObject = buildNotification(message);
+	function addMessageToQueue(queueName, message, messageTTL, callback) {
+		if (typeof messageTTL === 'function') {
+			callback = messageTTL;
+			messageTTL = undefined;
+		}
+		const notificationObject = buildNotification(message, undefined, undefined, messageTTL);
 		const notificationLifeTimer = queueMessageLifeTimers[queueName];
 
 		if(typeof queues[queueName] === "undefined"){
 			return callback(new Error(`There is no queue called ${queueName}`));
 		}
 
 		queues[queueName].push(notificationObject);
-		
+
 		if (typeof storageFolderPath !== 'undefined') {
 			return notificationObject.timeout = setTimeout(function () {
 				//maybe we don't need to do this ... bur for safety reasons...
```

Comment:
I don't get the reason for the loop. In its current form the loop iterates over the methods of the Queue instance (maybe you intended for `for .. of` instead of `for in`? Also, should the loop return without calling the callback?